src/main/java/io/unlogged/AgentCommandExecutorImpl.java

```diff
@@ -483,23 +483,104 @@ public AgentCommandResponse injectMocks(AgentCommandRequest agentCommandRequest)
+					// get list of bean names
+					// reflection: String[] beanNames = applicationContext.getBeanDefinitionNames()
+					Class<?> applicationContextClass = Class.forName("org.springframework.context.ApplicationContext");
+					Method getBeanDefinitionNamesMethod = applicationContextClass.getMethod("getBeanDefinitionNames");
+					String[] beanNames = (String[]) getBeanDefinitionNamesMethod.invoke(applicationContext);
+
+					// get bean from beanName
+					List<Object> applicationBean = new ArrayList<>();
+					for (String beanName: beanNames) {
+						// reflection: Object bean = applicationContext.getBean(beanName)
+
+						Method getBeanMethod = applicationContextClass.getMethod("getBean", String.class);
+						Object bean = getBeanMethod.invoke(applicationContext, beanName);
+						applicationBean.add(bean);
+					}
+
+					// inject mocks
+					// TODO: improve search performance here
+					Class<? extends Object> classDefination = null;
+					for (Object bean: applicationBean) {
+						if (classDefination == null ) {
+							Class<? extends Object> baseBeanClass = bean.getClass();
+							Class<? extends Object> beanClass = baseBeanClass;
+
+							while (beanClass != Object.class) {
+								if (sourceClassName.equals(beanClass.getName())) {
+									classDefination = baseBeanClass;
+									break;
+								}
+								beanClass = beanClass.getSuperclass();
+							}
+						}
+					}
+					classObject = classDefination;
+
+					// get a instance for classObject class
+					Constructor<?>[] constructors = classDefination.getDeclaredConstructors();
+					// check for a zero-args constructor
+					for (Constructor<?> constructor : constructors) {
+						if (constructor.getParameterCount() == 0) {
+							constructor.setAccessible(true);
+							sourceClassInstance = constructor.newInstance();
+						}
+					}
+
+					// If there is no zero-args constructor then build an instance from first constructor
+					if (sourceClassInstance == null) {
+						if (constructors.length > 0) {
+							Constructor<?> constructor = constructors[0];
+							// Define argument for the constructor (null for reference types, 0 for primitive types)
+							Object[] argsArray = new Object[constructor.getParameterCount()];
+							constructor.setAccessible(true);
+							sourceClassInstance = constructor.newInstance(argsArray);
+						} else {
+							// The class does not have any constructor
+							continue;
+						}
+					}
+				}
+				else {
+					// no instance found for this class
+					// nothing to inject mocks to
+					continue;
+				}
             }
             List<DeclaredMock> declaredMocks = mocksBySourceClass.get(sourceClassName);
 
             Map<String, List<DeclaredMock>> mocksByField = declaredMocks.stream()
                     .collect(Collectors.groupingBy(DeclaredMock::getFieldName));
 
-            Class<? extends Object> classObject = sourceClassInstance.getClass();
+			if (classObject == null) {
```

Comment:
@artpar If both the strategies give a classObject then the first strategy will take precedence. Is that the right priority order, or should the priority be give to the value that comes from logger?